src/v/coproc/event_listener.cc

```diff
@@ -252,4 +287,31 @@ event_listener::poll_topic(model::record_batch_reader::data_t& events) {
                                  : ss::stop_iteration::no;
 };
 
+ss::future<> event_listener::advertise_state_update(std::size_t n) {
```

Comment:
how often is this invoked, is there any pathological cases where it would be too much? should it be debounced or timer controlled (e.g. every 30 seconds if a change has been made) etc..?

Reply:
It is not in a hot path, its invoked everytime deploy, or remove is called, or after a recovery phase. It would be nice to have some behavior that disreguards old outstanding requests if a new request arises. Right now I just publish, and let compaction remove the duplicate.

Reply:
im not really asking if it is in a hot path, I'm asking if it will be invoked too frequently. at least the way I see it being used it can be run any time and it will do the correct thing. so time based updates made sense if there was an issue with load.

Reply:
It shouldn't be invoked too frequently. It will only be invoked on rpk wasm deploy/remove or when wasm state reconcile is occurring